docs/recipes/stencil.md

```diff
@@ -0,0 +1,79 @@
+# Usage with Stencil
+
+[Stencil](https://stenciljs.com/) web components work very well with XState.
+
+### `src/helpers/toggle-machine.ts`
+
+```js
+import { createMachine } from "@xstate/fsm";
+
+export const toggleMachine = createMachine<{ toggle: "active" | "inactive" }>({

```

Comment:
The first generic type parameter is actually for the context which is not used at all here, right?

I tried

```ts
type ToggleState = { context: {} } & ({ value: "active" } | { value: "inactive" });

export const toggleMachine = createMachine<{}, { type: "toggle" }, ToggleState>(/* ... */);
```

which I guess is correct type-wise(?), but it doesn't really give much type safety 😐 `this.state.value` is still inferred as `string` rather than the string literal type, and `send` is supposed to have type safety when you use a different syntax `send({ type: 'toggle' })` but it also just infers `type` as string for me 🤷🏻‍♂️  Maybe that only works with `xstate` but not `@xstate/fsm`?
(https://xstate.js.org/docs/guides/typescript.html#actions)

Reply:
Hm never mind I think it was some caching issue with codesandbox 🙈

It does infer the `send` type but `state.value` is inferred as `string`.